sdk/core/azure-core/inc/azure/core/io/body_stream.hpp

```diff
@@ -282,4 +284,45 @@ namespace Azure { namespace Core { namespace IO {
     int64_t Length() const override;
   };
 
+  /**
+   * @brief A concrete implementation of #Azure::Core::IO::BodyStream that wraps another stream and
+   * reports progress
+   */
+  class ProgressBodyStream : public BodyStream {
+  private:
+    BodyStream* m_bodyStream;
+    std::atomic_int64_t m_position;
+    std::function<void(int64_t position, int64_t length)> m_callback;
```

Comment:
The callback should not include the length as this may not be known ahead of time.
I think 'position' is better named 'bytesTransferred'.

Reply:
done